File: code_snippets/java/src/main/java/develop/clients/GreeterService.java
package develop.clients;

import dev.restate.sdk.Context;
import dev.restate.sdk.annotation.Handler;
import dev.restate.sdk.annotation.Service;

@Service
public class GreeterService {
  @Handler
  public String greet(Context ctx, String greeting) {
    return "Hello, " + greeting + "!";
  }
}

File: code_snippets/kotlin/src/main/kotlin/develop/clients/GreeterService.kt
package develop.clients

import dev.restate.sdk.annotation.Handler
import dev.restate.sdk.annotation.Service
import dev.restate.sdk.kotlin.Context

@Service
class GreeterService {
  @Handler
  suspend fun greet(ctx: Context, greeting: String): String {
    return "Hello, $greeting!"
  }
}
